File: javascript/src/euler/solution005.js
// https://projecteuler.net/problem=5

// 2520 is the smallest number that can be divided by each of the numbers from
// 1 to 10 without any remainder.
// What is the smallest positive number that is evenly divisible by all of the
// numbers from 1 to 20?

// Answer: 232792560

module.exports = {
  answer : () => {
    var answer = 0;
    var test = 20;
    var check = false;

    while( !check ) {
      check = true;
      for( i = 20 ; i && check ; --i ) {
        check &= (0 == (test % i));
      }
      if( !check ) {
        test += 20;
      }
    }
    return test;
  }
};

File: java/src/main/java/org/tvarley/euler/solutions/Solution005.java
/*
 * Solution for Euler problem 005
 *
 * Smallest multiple
 *
 * 2520 is the smallest number that can be divided by each of the numbers from
 * 1 to 10 without any remainder.
 *
 * What is the smallest positive number that is evenly divisible by all of the
 * numbers from 1 to 20?
 *
 * @author tvarley
 * @date 3/8/2026
 */
package org.tvarley.euler.solutions;

import org.tvarley.euler.Solution;

public class Solution005 implements Solution {
  public String solve() {
    int lcm = 1;
    for (int i = 2; i <= 20; i++) {
      lcm = lcm(lcm, i);
    }
    return Integer.toString(lcm);
  }

  private int lcm(int a, int b) {
    return a * (b / gcd(a, b));
  }

  private int gcd(int a, int b) {
    while (b != 0) {
      int t = b;
      b = a % b;
      a = t;
    }
    return a;
  }
}